### PR TITLE
Add HuggingFace tokenizer as an Evaluator

### PR DESCRIPTION
**Pull request by co42**
Le binding Rust se trouve dans `evaluator-huggingface/huggingface-tokenizer-jni`

### CODE REVIEW

File: api/src/test/java/com/ovh/mls/serving/runtime/huggingface/HuggingFaceTokenizerTest.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import java.util.Map;
+
+import static com.jayway.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.equalTo;
+
+public class HuggingFaceTokenizerTest {
+
+    @BeforeAll
+    public static void startServer() throws NoSuchFieldException, IllegalAccessException {
+        // Reset swagger context
```
Comment (christophe-rannou) [inline api/src/test/java/com/ovh/mls/serving/runtime/huggingface/HuggingFaceTokenizerTest.java:25]: Why is it necessary to reset the swagger ?

File: commons/src/main/java/com/ovh/mls/serving/runtime/utils/NativeUtils.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.ovh.mls.serving.runtime.utils;
+
+import java.io.*;
```
Comment (christophe-rannou) [inline commons/src/main/java/com/ovh/mls/serving/runtime/utils/NativeUtils.java:26]: If @XciD sees this he will hate you, he seems to despise anybody who dares adding an `*` import

File: commons/src/main/java/com/ovh/mls/serving/runtime/utils/NativeUtils.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.ovh.mls.serving.runtime.utils;
```
Comment (XciD) [inline commons/src/main/java/com/ovh/mls/serving/runtime/utils/NativeUtils.java:24]: you could add the link to the original class.

File: evaluator-huggingface/src/main/java/com/ovh/mls/serving/runtime/huggingface/tokenizer/HuggingFaceTokenizerEvaluator.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+    private final Tokenizer tokenizer;
+
+    @Override
+    public TensorIO evaluate(TensorIO io, EvaluationContext evaluationContext) throws EvaluationException {
+        String input = (String) io.getTensors().get("sequence").getCoord(0);
+        Encoding encoding = tokenizer.encode(input);
+
+        TensorIO output = new TensorIO();
+        output.getTensors().put("tokens", new Tensor(
```
Comment (XciD) [inline evaluator-huggingface/src/main/java/com/ovh/mls/serving/runtime/huggingface/tokenizer/HuggingFaceTokenizerEvaluator.java:23]: Can you extract name in constant ?

File: api/src/test/java/com/ovh/mls/serving/runtime/huggingface/HuggingFaceTokenizerTest.java (hunk 1 above)
Comment (co42) [reply api/src/test/java/com/ovh/mls/serving/runtime/huggingface/HuggingFaceTokenizerTest.java:25]: I'm not sure I copy-pasted it from an other unit test. Also the Api port is always different. @XciD do you know why ?

File: commons/src/main/java/com/ovh/mls/serving/runtime/utils/NativeUtils.java (hunk 3 above)
Comment (co42) [reply commons/src/main/java/com/ovh/mls/serving/runtime/utils/NativeUtils.java:24]: There is a link line 40

File: api/src/test/java/com/ovh/mls/serving/runtime/huggingface/HuggingFaceTokenizerTest.java (hunk 1 above)
Comment (XciD) [reply api/src/test/java/com/ovh/mls/serving/runtime/huggingface/HuggingFaceTokenizerTest.java:25]: You need to reset the swagger in order to re-gen it. Don't needed on this case I think.
And about the port, you are right, you need a random one if tests runs in parallel

**Pull request merged**